crates/swc_ecma_lints/src/rules/dot_notation.rs

```diff
@@ -0,0 +1,143 @@
+}
+
+impl Debug for DotNotation {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("DotNotation")
+            .field("expected_reaction", &self.expected_reaction)
+            .field("allow_keywords", &self.allow_keywords)
+            .field("pattern", &self.pattern)
+            .field("is_module", &self.is_module)
+            .finish()
+    }
+}
+
+impl DotNotation {
+    fn new(
+        source_map: Arc<SourceMap>,
+        is_module: bool,
+        config: &RuleConfig<DotNotationConfig>,
+    ) -> Self {
+        let dot_notation_config = config.get_rule_config();
+
+        Self {
+            expected_reaction: *config.get_rule_reaction(),
+            allow_keywords: dot_notation_config.allow_keywords.unwrap_or(true),
+            source_map,
+            is_module,
+            pattern: dot_notation_config.allow_pattern.clone(),
+        }
+    }
+
+    fn emit_report(&self, span: Span, quote_type: QuotesType, prop: &str) {
+        let message = format!(
+            "[{quote}{prop}{quote}] is better written in dot notation",
+            prop = prop,
+            quote = quote_type.get_char()
+        );
+
+        HANDLER.with(|handler| match self.expected_reaction {
+            LintRuleReaction::Error => {
+                handler.struct_span_err(span, &message).emit();
+            }
+            LintRuleReaction::Warning => {
+                handler.struct_span_warn(span, &message).emit();
+            }
+            _ => {}
+        });
+    }
+
+    fn check(&self, span: Span, quote_type: QuotesType, prop_name: &str) {
+        if self.allow_keywords
+            && (prop_name.is_reserved() || prop_name.is_reserved_in_strict_mode(self.is_module))
+        {
+            return;
+        }
+
+        if let Some(pattern) = &self.pattern {
+            static REGEX_CACHE: Lazy<DashMap<String, Regex, ahash::RandomState>> =
+                Lazy::new(Default::default);
+
+            if !REGEX_CACHE.contains_key(pattern) {
+                REGEX_CACHE.insert(
+                    pattern.clone(),
+                    Regex::new(pattern).expect(INVALID_REGEX_MESSAGE),
+                );
+            }
+
+            if REGEX_CACHE.get(pattern).unwrap().is_match(prop_name) {
+                return;
+            }
+        }
+
+        self.emit_report(span, quote_type, prop_name);
+    }
+}
+
+impl Visit for DotNotation {
+    noop_visit_type!();
+
+    fn visit_member_prop(&mut self, member: &MemberProp) {
+        if let MemberProp::Computed(prop) = member {
+            match &*prop.expr {
+                Expr::Lit(Lit::Str(lit_str)) => {
+                    let quote_type = resolve_string_quote_type(&self.source_map, lit_str).unwrap();
+
+                    self.check(prop.span, quote_type, &*lit_str.value);
+                }
+                Expr::Member(member) => {
+                    member.visit_children_with(self);
+                }
+                _ => {}
```

Comment:
I think we need `prop.visit_with(self)` is required.

Reply:
Why?

Reply:
Because `foo[call(my['var'])]` will not be handled otherwise.

Reply:
Oh.. I fixed it and add your example as test case